src/authentication/steps/SelectServer.jsx

```diff
@@ -21,81 +22,92 @@ export class SelectServer extends Component {
   }
 
   componentDidMount() {
-    this.serverInput.focus()
+    this.input.focus()
   }
 
   componentWillReceiveProps(nextProps) {
-    this.setState({
+    this.setState(state => ({
+      ...state,
```

Comment:
Why do you use a setState callback here ?

Reply:
I always use `setState` in its functional form.
See https://medium.freecodecamp.org/functional-setstate-is-the-future-of-react-374f30401b6b
And https://twitter.com/dan_abramov/status/824308413559668744

Reply:
Phew such a long article for a simple information :D TLDR : if your state update relies on the previous state, use a functional setState since setState applies its updates asynchronously; by doing `setState(updates)`, you can have a race condition.

From the tweet, the advantage I see is from a testing point of view since you can test each individual test updater.

Understood, thanks for the pointers.

Reply:
And a conclusion: always use the functional `setState` as it does not cost much more and it is a very good and cleaner code with pure functions and race condition handled.